### PR TITLE
Add element-selector to group block

### PR DESCRIPTION
**Pull request by aristath**
## Description

This is a continuation of https://github.com/WordPress/gutenberg/pull/27101
In that PR we added an element selector to the template-part block. This PR copies the same logic to group blocks.
Came up in https://github.com/WordPress/theme-experiments/issues/180 where it became evident that the `<main>` element was missing in the theme. Though editing it in templates is the preferred way to handle things for theme-included templates, there's currently no way to do the same for user-created templates.

## How has this been tested?

Tested with existing group blocks and verified nothing breaks, there's no backwards-compatibility issues.
Tested with new group blocks and verified that the tag gets properly applied, saved and presented.

## Screenshots 

The control was added to the Advanced section - just like in the template-part block.

Before | After
-------- | --------
Advanced section screenshot without the HTML Element control | Advanced section screenshot with the HTML Element control

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] ~~I've included developer documentation if appropriate.~~ 
- [ ] ~~I've updated all React Native files affected by any refactorings/renamings in this PR.~~

### CODE REVIEW

**Comment by carolinan** [comment]
Tested with TT1 Blocks, works well.

File: packages/block-library/src/group/edit.js [hunk 1]
```diff
@@ -33,14 +38,36 @@ function GroupEdit( { attributes, clientId } ) {
 		}
 	);
 
-	return (
-		<TagName { ...blockProps }>
-			<BoxControlVisualizer
-				values={ attributes.style?.spacing?.padding }
-				showValues={ attributes.style?.visualizers?.padding }
+	const inspectorAdvancedControls = (
+		<InspectorAdvancedControls>
```
Comment (mtias) [inline packages/block-library/src/group/edit.js:42]: Any reason for not inlining this in the main `return` call?

Comment (aristath) [reply packages/block-library/src/group/edit.js]: No reason other than consistency with the template-part block where we do the same.
I inlined it now, and did the same for the template-part :+1:

**Comment by gziolo** [comment]
@aristath, any idea why so many e2e tests related to FSE fail with this branch?

The change proposed to align with the Group block makes sense. I like that it's inside the advanced section since it's very technical but shares the same characteristics as the class name or anchor.

**Comment by aristath** [comment]
> @aristath, any idea why so many e2e tests related to FSE fail with this branch?

It doesn't seem to be related to this PR... I can see the same tests are currently failing on the `master` branch :thinking:

**Review by mtias — APPROVED** [review]

**Pull request merged**